### DISCUSSION TITLE
Definition of opposition?

### DISCUSSION THREAD

**Discussion by Atque** [Stellarium/stellarium#1894] [Q&A] [answered]
I'm having troubles understanding how AstroCalc defines an opposition. Calculating the Mars opposition of 2022, it is said to occur exactly on 2022-12-08 04:38:59 UTC (btw the Moon will be very close to Mars, even occulting it from some places!) The screenshot below shows the geocentric view of this event.
![stellarium-109](https://user-images.githubusercontent.com/37028368/133883362-70d21fc8-cddf-41a3-bca3-329daad04a9b.png)

What I don't fully understand is why Mars is not exactly on the O/C line. Why is that? Wouldn't an opposition mean that an object is exactly on the O/C line, or is there some other criterion?

Geocentric view, latest beta (with aberration), DE440 ephemeris.

**Comment by gzotti** [comment, marked as answer]
Wow, that bloody moon will kill my Mars opposition surge observation! :-) :-) :-)
Opposition is nowadays mostly defined in RA, not ecliptical longitude (which the OC line shows). There is still a small difference, likely some algorithmic abbreviation which delivers a "good enough" solution. Don't trust the minutes...

**Reply by Atque** [reply]
Interestingly, the 2018 opposition of Mars also took place on the night of a full moon, but thankfully, the Moon was then eclipsed.

Aha, so it's defined by RA. Still it's off, but it's off in SkySafari too (SkySafari claims it happens about 1 minute earlier, but that does not make up for the offset in RA).

**Reply by gzotti** [reply]
I remember that memorable eclipse... ![image](https://user-images.githubusercontent.com/18099873/133888108-2fa38108-a188-4ca5-84dc-745e0fee3d38.png)
Now we need an eclipse which covers Mars (or another planet) in opposition...

**Reply by Atque** [reply]
Wow, that's a nice shot. Did you take it? It was amazing, even from my northern latitude.

Lunar eclipse combined with the occultation of a planet? Does that even happen once in a millennium? :)

**Reply by gzotti** [reply]
Sure. This was towards the end of the eclipse, and the flashes fired pretty rapidly. 
Eclipse with occultation: Something in Meeus' Morsels?

**Reply by Atque** [reply]
I did a quick Google search but could not find anything. It might be found in some printed book.
